src/Orleans/Configuration/NodeConfiguration.cs

```diff
@@ -489,6 +494,9 @@ internal void Load(XmlElement root)
                             StartupTypeName = child.GetAttribute("Type");
                         }
                         break;
+                    case "AssemblyCatalog":
+                        Assemblies = ConfigUtilities.ParseAssemblyList(child);
+                        break;
```

Comment:
Feels odd to have this return the list of assemblies, and then separately have the runtime read the list to create the catalog, and having 2 mutually exclusive properties to assign